sdk/python/kfp/cli/components.py

```diff
@@ -0,0 +1,382 @@
+            _info('Building KFP package from local directory {}'.format(
+                typer.style(str(kfp_package_path), fg=typer.colors.CYAN)))
+            temp_dir = pathlib.Path(tempfile.mkdtemp())
+            try:
+                subprocess.run([
+                    'python3',
+                    kfp_package_path / 'setup.py',
+                    'bdist_wheel',
+                    '--dist-dir',
+                    str(temp_dir),
+                ],
+                               cwd=kfp_package_path)
+                wheel_files = list(temp_dir.glob('*.whl'))
+                if len(wheel_files) != 1:
+                    _error('Failed to find built KFP wheel under {}'.format(
+                        temp_dir))
+                    raise typer.Exit(1)
+
+                wheel_file = wheel_files[0]
+                shutil.copy(wheel_file, self._context_directory)
+                self._kfp_package_path = wheel_file.name
+                self._maybe_copy_kfp_package = 'COPY {wheel_name} {wheel_name}'.format(
+                    wheel_name=self._kfp_package_path)
+            except subprocess.CalledProcessError as e:
+                _error('Failed to build KFP wheel locally:\n{}'.format(e))
+                raise typer.Exit(1)
+            finally:
+                _info('Cleaning up temporary directory {}'.format(temp_dir))
+                shutil.rmtree(temp_dir)
+        else:
+            self._kfp_package_path = kfp_package_path
+
+        _info('Building component using KFP package path: {}'.format(
+            typer.style(str(self._kfp_package_path), fg=typer.colors.CYAN)))
+
+        self._files = [
+            file.name
+            for file in self._context_directory.glob('*')
+            if file.is_file()
+        ]
+
+        self._base_image = None
+        self._target_image = None
+        self._load_components()
+
+    def _load_components(self):
+        module_components: component_factory.ModuleComponents = {}
+        for component_module in self._component_modules:
+            module_name = component_module.name[:-len('.py')]
+            module_directory = component_module.parent
+            utils.load_module(module_name=module_name,
+                              module_directory=module_directory)
+
+            formatted_module_file = typer.style(str(self._component_modules),
+                                                fg=typer.colors.CYAN)
+
+            module_components.update(
+                component_factory.REGISTERED_MODULE_COMPONENTS.get(
+                    component_module, None))
+
+            if module_components is None:
+                _error('Failed to load components from module file {}'.format(
+                    formatted_module_file))
+                raise typer.Exit(1)
+
+            if not module_components:
+                _error('No KFP components found in file {}'.format(
+                    formatted_module_file))
+                raise typer.Exit(1)
```

Comment:
What if we just ignore this here? Would we be able to build the entire folder into the container then?
One CUJ is users may have their custom helper code defined in some utils files.

I'm trying to see where's the challenge to support an interface `kfp component build <folder>` and we scan for kfp components within the folder but package the entire folder into the container as is.

Reply:
We wouldn't be able to figure out the entrypoint though, and users would have to handwrite the YAML file. Also, many assumptions like the presence of `base_image` and `target_image` would break. I think we can provide a separate command for this option, which will make that clearer. WDYT?